flocker/node/agents/ebs.py

```diff
@@ -590,6 +602,11 @@ def destroy_volume(self, blockdevice_id):
                         'Failed to delete volume: {!r}'.format(blockdevice_id)
                     )
                 else:
+                    ebs_volume = self._get_ebs_volume(blockdevice_id)
```

Comment:
Since we just deleted the volume, it seems plausible (though significantly unlikely) that this could fail with `UnknownVolume`.  It just depends whether EC2 manages to destroy the volume faster than we manage to make the `connection.get_all_volumes` call.

I have no realistic suggestions for how to test this case but it seems like it could be avoided by making the `_get_ebs_volume` call before the `connection.delete_volume` call.